File: CS/Dashboard_DataLoading/Form1.cs
using System;
using System.Data;
using DevExpress.XtraEditors;
using DevExpress.DashboardCommon;

namespace Dashboard_DataLoading {
    public partial class Form1 : XtraForm {
        public Form1() {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e) {
            Dashboard dashboard = new Dashboard();
            DashboardObjectDataSource objectDataSource = new DashboardObjectDataSource();
            objectDataSource.DataSource = GetData();
            dashboard.DataSources.Add(objectDataSource);
            dashboardViewer1.Dashboard = dashboard; 

            // Creates a new grid dashboard item with two columns that display car models and prices.
            GridDashboardItem grid1 = new GridDashboardItem();
            grid1.DataSource = dashboard.DataSources[0];
            grid1.Columns.Add(new GridDimensionColumn(new Dimension("Model")));
            grid1.Columns.Add(new GridMeasureColumn(new Measure("Price")));
            dashboard.Items.Add(grid1);                       
        }

        // Handles the DashboardViewer.DataLoading event to provide the dashboard with new data.
        private void dashboardViewer1_DataLoading(object sender, DataLoadingEventArgs e) {
            if (e.DataSourceName == "Object Data Source 1") {
                e.Data = UpdateData();
            }
        }

        public DataTable GetData() {
            DataSet xmlDataSet = new DataSet();
            xmlDataSet.ReadXml(@"..\..\Data\Cars_1.xml");
            return xmlDataSet.Tables["Cars"];
        }

        public DataTable UpdateData() {
            DataSet xmlDataSet = new DataSet();
            xmlDataSet.ReadXml(@"..\..\Data\Cars_2.xml");
            return xmlDataSet.Tables["Cars"];
        }

        private void button1_Click(object sender, EventArgs e) {
            // Reloads data in data sources.
            dashboardViewer1.ReloadData();
        }
    }
}

File: VB/Dashboard_DataLoading/Form1.vb
Imports System
Imports System.Data
Imports DevExpress.XtraEditors
Imports DevExpress.DashboardCommon

Namespace Dashboard_DataLoading

    Public Partial Class Form1
        Inherits XtraForm

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs)
            ' Specifies a new dashboard as a currently opened dashboard.
            dashboardViewer1.Dashboard = New Dashboard()
            ' Creates a data source for the dashboard.
            dashboardViewer1.Dashboard.AddDataSource("Data Source 1", GetData())
            ' Creates a new grid dashboard item with two columns that display car models and prices.
            Dim grid1 As GridDashboardItem = New GridDashboardItem()
            grid1.DataSource = dashboardViewer1.Dashboard.DataSources(0)
            grid1.Columns.Add(New GridDimensionColumn(New Dimension("Model")))
            grid1.Columns.Add(New GridMeasureColumn(New Measure("Price")))
            dashboardViewer1.Dashboard.Items.Add(grid1)
        End Sub

        ' Handles the DashboardViewer.DataLoading event to provide the dashboard with new data.
        Private Sub dashboardViewer1_DataLoading(ByVal sender As Object, ByVal e As DevExpress.DataAccess.DataLoadingEventArgs)
            If Equals(e.DataSourceName, "Data Source 1") Then
                e.Data = UpdateData()
            End If
        End Sub

        Public Function GetData() As DataTable
            Dim xmlDataSet As DataSet = New DataSet()
            xmlDataSet.ReadXml("..\..\Data\Cars_1.xml")
            Return xmlDataSet.Tables("Cars")
        End Function

        Public Function UpdateData() As DataTable
            Dim xmlDataSet As DataSet = New DataSet()
            xmlDataSet.ReadXml("..\..\Data\Cars_2.xml")
            Return xmlDataSet.Tables("Cars")
        End Function

        Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs)
            ' Reloads data in data sources.
            dashboardViewer1.ReloadData()
        End Sub
    End Class
End Namespace
